forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -206,18 +263,89 @@ pub async fn deploy(command: cmd::Deploy) -> Result<Vec<DeployedContract>> {
                     bail!("Both `--salt` and `--default-salt` were specified: must choose one")
                 }
             };
-            let contract_id = deploy_pkg(
+            let node_url = get_node_url(&command.node, &pkg.descriptor.manifest_file.network)?;
+            info!(
+                "  {} contract: {}",
+                "Deploying".bold().green(),
+                &pkg.descriptor.name
+            );
+            let deployed_contract_id = deploy_pkg(
                 &command,
                 &pkg.descriptor.manifest_file,
                 &pkg,
                 salt,
                 &wallet_mode,
             )
             .await?;
-            contract_ids.push(contract_id);
+            let proxy = &pkg.descriptor.manifest_file.proxy();
+            let proxy_id = if let Some(proxy) = proxy {
+                if proxy.enabled {
+                    if let Some(proxy_addr) = &proxy.address {
+                        // Make a call into the contract to update impl contract address to 'deployed_contract'.
+
+                        // Create a contract instance for the proxy contract using default proxy contract abi and
+                        // specified address.
+                        info!("  {} proxy contract", "Updating".bold().green());
+                        let provider = Provider::connect(node_url.clone()).await?;
+                        // TODO: once https://github.com/FuelLabs/sway/issues/6071 is closed, this will return just a result
+                        // and we won't need to handle the manual prompt based signature case.
+                        let signing_key = select_secret_key(
+                            &wallet_mode,
+                            command.default_signer,
+                            command.signing_key,
+                            &provider,
+                        )
+                        .await?;
+
+                        let signing_key = signing_key.ok_or_else(
+
+                            || anyhow::anyhow!("proxy contract deployments are not supported with manual prompt based signing")
+                        )?;
+                        let proxy_contract =
+                            ContractId::from_str(proxy_addr).map_err(|e| anyhow::anyhow!(e))?;
+
+                        update_proxy_contract_target(
+                            provider,
+                            signing_key,
+                            proxy_contract,
+                            deployed_contract_id,
+                        )
+                        .await?;
+                        Some(proxy_contract)
+                    } else {
+                        // Deploy a new proxy contract.
+                        let deployed_proxy_contract = deploy_new_proxy(
+                            &pkg,
+                            &mut owner_account_address,
```

Comment:
It looks like `owner_account_address` never gets set to anything other than the default.